extensions/amp-story/1.0/amp-story.js

```diff
@@ -1201,21 +1204,30 @@ export class AmpStory extends AMP.BaseElement {
    * @private
    */
   updateStoryNavigationPath_(targetPageId, direction) {
-    const targetPage = this.getPageById(targetPageId);
     if (direction === NavigationDirection.PREVIOUS) {
       this.storyNavigationPath_.pop();
       if (this.storyNavigationPath_.length > 0) {
         const pathPrevious =
-            this.storyNavigationPath_[this.storyNavigationPath_.length - 1] ||
-            this.storyNavigationPath_[0];
-        if (pathPrevious.element.id != targetPageId) {
-          return pathPrevious;
+            this.storyNavigationPath_[this.storyNavigationPath_.length - 1];
+        if (pathPrevious != targetPageId) {
+          return this.getPageById(pathPrevious);
         }
       }
     } else if (direction === NavigationDirection.NEXT) {
-      this.storyNavigationPath_.push(targetPage);
+      const topOfStack =
+          this.storyNavigationPath_[this.storyNavigationPath_.length - 1];
+      // If the user navigates the away from the page, the top of storyStack
+      // will be the same as ampStoryPageId in the history state.
+      if ((targetPageId !== topOfStack) ||
+        this.storyNavigationPath_.length === 0) {
```

Comment:
It looks like if `this.storyNavigationPath_.length` is 0, `topOfStack` will be `undefined`.
So `targetPageId !== topOfStack` will always be `true` for `this.storyNavigationPath_.length` of 0.

Can we replace this if with just the first expression `if (targetPageId !== topOfStack)`? Lmk if I'm missing something!

Reply:
Done